mcnp/mcnp_funcs.cpp

```diff
@@ -498,37 +520,53 @@ void write_lcad_old(std::ofstream &lcadfile)
       get_real_prop( vol, cellid, "imp.e", imp_e );
     }
 
-    lcadfile << cellid << " ";
+    // If no importances specified, default to neutron mode
+    if ( ! imp_n_needed && ! imp_p_needed && ! imp_e_needed ) {
+      imp_n_needed = true;
+      cimp_n = 1;
+      imp_n = 1;
+    }
+
+    lcadfile << cellid;
 
     bool graveyard = DAG->has_prop( vol, "graveyard" );
 
     if( graveyard ) {
-      lcadfile << " 0 imp:n=0";
+      lcadfile << " 0";
+      if( imp_n_needed ) lcadfile << " imp:n=0";
+      if( imp_p_needed ) lcadfile << " imp:p=0";
+      if( imp_e_needed ) lcadfile << " imp:e=0";
+      lcadfile << " $ graveyard";
       if( DAG->has_prop(vol, "comp") ) {
         // material for the implicit complement has been specified.
         get_int_prop( vol, cellid, "mat", cmat );
         get_real_prop( vol, cellid, "rho", crho );
-        std::cout << "Detected material and density specified for implicit complement: " << cmat <<", " << crho << std::endl;
-        cimp = imp_n;
+        std::cout << "Detected material and density specified for implicit complement: " << cmat << ", " << crho << std::endl;
+        if( imp_n_needed ) cimp_n = imp_n;
```

Comment:
Doesn't this change the "default" values that are reused in line 508 in another iteration of the loop?